camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelResolver.java

```diff
@@ -188,50 +204,123 @@ Collection<File> resolve(String organization, String name, String revision, Stri
         DefaultModuleDescriptor md =
                 DefaultModuleDescriptor.newDefaultInstance(ModuleRevisionId.newInstance(organization,
                                                         name + "-caller", "working"));
-        
+
         DefaultDependencyDescriptor dd = new DefaultDependencyDescriptor(md,
                                                                          ModuleRevisionId.newInstance(organization, name, revision),
                                                                          false,
                                                                          false,
                                                                          true);
-        dd.addDependencyConfiguration("default", "default");
-        
+        dd.addDependencyConfiguration(DEFAULT_CONFIGURATION, ANY_CONFIGURATION);
+
         if (type != null) {
-            // TODO: Figure out why both of these are necessary, but it works when present & not when not.
             Artifact typedArtifact = new DefaultArtifact(ModuleRevisionId.newInstance(organization, name, revision),
                                                          null, name, type, "." + type);
+            log.debug(">>>> Specifying artifact for {}:{}", name, type);
             DependencyArtifactDescriptor dad =
                     new DefaultDependencyArtifactDescriptor(dd, name, type, type, null, null);
             dd.addDependencyArtifact(DEFAULT_CONFIGURATION, dad);
             md.addArtifact(DEFAULT_CONFIGURATION, typedArtifact);
         }
         md.addDependency(dd);
         
-        log.trace("Resolving required libraries -- {}", purpose);
-        //init resolve report
+        // There are some dependencies that don't seem to resolve correctly, at least with our implementation.  To
+        // counter this, we track a set of places where we need to override the type provided by the maven repo.
+        // This situation typically arises when the artifact in question specifies a packaging attribute for some
+        // configurations, and the configuration for which we're locking qualifies.  To manage this, we'll add a
+        // DependencyDescriptorMediator (via the lambda below) which, when it sees the dependency specified in the
+        // typeOverrides list, will add a DependencyArtifact to force the downloading of th correct type/extension.
+        // these adjustments allow us to include having to special case lots of stuff in the kamelet/assembly
+        // specifications.  It also will provide the same functions if users configure their own camelConnectors that
+        // encounter these situations.
+        
```

Comment:
for each dependency in our override list, add a mediator.  The mediator will add an artifact type descriptor so that the underlying ivy code knows what to go fetch.